templates/include/telemetry/mainPayload.1.schema.json

```diff
@@ -63,10 +63,19 @@
       "properties": { }
     },
     "log": {
-      "type": [
-        "array",
-        "object"
-        ]
+      "type": "array",
+      "items": {
+        "type": "array",
+        "items": {
+          "type": [
+            {"type": "string"},
+            {"type": "number"},
+            {"type": "string"},
+            {"type": "string"},
+            {"type": "string"}
+          ]
+        }
+      }
```

Comment:
I'm pretty uncomfortable with this change because I'm not sure how stable this field is. Based on sampled pings from 2019-03-14, they take the form of a tuple like above:

```
$ jq ".payload.log // empty | select(length != 0)" data/telemetry.main.4.ndjson
[
  [
    "EXPERIMENT_ACTIVATION",
    17077,
    "REJECTED",
    "unaccelerated-compositor-process-nightly56@experiments.mozilla.org",
    "endTime"
  ]
]
[
  [
    "EXPERIMENT_ACTIVATION",
    2293,
    "REJECTED",
    "unaccelerated-compositor-process-nightly56@experiments.mozilla.org",
    "endTime"
  ]
]
[
  [
    "EXPERIMENT_ACTIVATION",
    13025,
    "REJECTED",
    "unaccelerated-compositor-process-nightly56@experiments.mozilla.org",
    "endTime"
  ],
  [
    "EXPERIMENT_ACTIVATION",
    3606165,
    "REJECTED",
    "unaccelerated-compositor-process-nightly56@experiments.mozilla.org",
    "endTime"
  ]
]
...
```

@mreid-moz do you have more context around this particular field?

Reply:
I don't know much about this field - @badboy do you know where this is recorded in the Firefox source?

Reply:
That's .... an [empty array](https://searchfox.org/mozilla-central/rev/7abb9117c8500ed20833746c9f8e800fce3a4688/toolkit/components/telemetry/pings/TelemetrySession.jsm#897) now since I removed it [about exactly a year ago](https://bugzilla.mozilla.org/show_bug.cgi?id=1443614). 

I don't actually know when we removed the collection code, so if you really need that info I can track it down.

Reply:
I'm curious about the `LOG_ENTRY_MAX_COUNT` variable in the patch to determine if this is actually a tuple or a variable length list with differing types.

Depending on how this is defined, we could get a different Avro type in the end (e.g. `"additionaltems": "string"` for an Array<Array<String>> vs `"additionaltems": false` for an Array<Struct>). Alternatively, removing the log field would be more reflective of our current schema at the expense of dropping the values.

In either case, I wouldn't want pings from older revisions to be dropped due to the validation schema being too strict.

Reply:
[Here's the old code](https://github.com/mozilla/gecko-dev/commit/201d19b399bd63af762bc6c8e1fa29a252bdc4e8#diff-3439f9c7661275a769f1816023db8931) of `TelemetryLog`.
The size of the outer array is limited by `LOG_ENTRY_MAX_COUNT`, but the entries itself are pretty much any array of the form: `[String, Number, String*]` (so 2 or more elements), but not a struct.

Reply:
I'm going to introduce the proper schema for this in a separate PR, since it's doing a slightly different thing.